client.go

```diff
@@ -170,3 +171,92 @@ func Exec(ctx context.Context, cmd string, args ...interface{}) error {
 func Query(ctx context.Context, cmd string, args ...interface{}) Args {
 	return DefaultClient.Query(ctx, cmd, args...)
 }
+
+type MultiError []error
+
+func (err MultiError) Error() string {
+	if err != nil && len(err) > 0 {
+		return fmt.Sprintf("%d errors. First one: '%s'.", len(err), err[0])
+	}
+	return fmt.Sprintf("%v", nil)
+}
+
+// WriteTestPattern writes a test pattern to a Redis client. The objective is to read the test pattern back
+// at a later stage using ReadTestPattern.
+func WriteTestPattern(client *Client, n int, keyTempl string, timeout time.Duration) (numSuccess int, numFailure int, err MultiError) {
+	writeErrs := make(chan error, n)
+	waiter := &sync.WaitGroup{}
+	for i := 0; i < n; i++ {
+		key := fmt.Sprintf(keyTempl, i)
+		val := "1"
+		waiter.Add(1)
+		go func(key, val string) {
+			defer waiter.Done()
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
+			defer cancel()
+			if err := client.Exec(ctx, "SET", key, val); err != nil {
+				writeErrs <- err
+			}
+		}(key, val)
+	}
+	waiter.Wait()
+	close(writeErrs)
+	numFailure = len(writeErrs)
+	numSuccess = n - numFailure
+	if len(writeErrs) > 0 {
+		err = make(MultiError, 0)
+		for writeErr := range writeErrs {
+			err = append(err, writeErr)
+		}
+	}
+	return
+}
+
+// ReadTestPattern reads a test pattern (previously writen using WriteTestPattern) from a Redis client and returns hit statistics.
+func ReadTestPattern(client *Client, n int, keyTempl string, timeout time.Duration) (numHits int, numMisses int, numErrors int, err MultiError) {
```

Comment:
If you put it in the `redistest` subpackage it's available from everywhere else, you'd call `redistest.ReadTestPattern` instead of `redis.ReadTestPattern` ;)